### ISSUE TITLE
SIM based vulnerabilities exploited by surveillance companies for espionage operation

### ISSUE THREAD

**Issue by CHEF-KOCH** [CHEF-KOCH/NSABlocklist#46]
The following basically explains why all this is possible:

* https://thehackernews.com/2019/09/simjacker-mobile-hacking.html
* https://simjacker.com/
* https://www.adaptivemobile.com/blog/simjacker-next-generation-spying-over-mobile
* https://indianexpress.com/article/technology/tech-news-technology/why-sim-swapping-is-a-growing-security-nightmare-5989743/

### Video Demo:
* https://player.vimeo.com/video/359352481

### Migration
SIM Alliance to help mitigate S@T-Push attacks
* https://simalliance.org/wp-content/uploads/2019/08/Security-guidelines-for-S@T-Push-v1.pdf

```
What makes this Attack work and why is it Special?

The attack relies both on these specific SMS messages being allowed, and 
the S@T Browser software being present on the UICC in the targeted 
phone. Specific SMS messages targeting UICC cards have been demonstrated 
before on how they could be exploited for malicious purposes. The 
Simjacker attack takes a different approach, and greatly simplifies and 
expands the attack by relying on the S@T Browser software as an 
execution environment. The S@T (pronounced sat) Browser – or SIMalliance 
Toolbox Browser  to give it its full name – is an application specified 
by the SIMalliance, and can be installed on a variety of UICC (SIM 
cards), including eSIMs. This S@T Browser software is not well known, is 
quite old, and its initial purpose was to enable services such as 
getting your account balance through the SIM card. Globally, its 
function has been mostly superseded by other technologies, and its 
specification has not been updated since 2009, however, like many legacy 
technologies it is still been used while remaining in the background. In 
this case we have observed the S@T protocol being used by mobile 
operators in at least 30 countries whose cumulative population adds up 
to over a billion people, so a sizable amount of people are potentially 
affected. It is also highly likely that additional countries have mobile 
operators that continue to use the technology on specific SIM cards.

This attack is also unique, in that the Simjacker Attack Message could 
logically be classified as carrying a complete malware payload, 
specifically spyware. This is because it contains a list of instructions 
that the SIM card is to execute. As software is essentially a list of 
instructions, and malware is ‘bad’ software, then this could make the 
Simjacker exploit the first real-life case of malware (specificially 
spyware) sent within a SMS. Previous malware sent by SMS - such as the 
incidents we profiled here - have involved sending links to malware, not 
the malware itself within a complete message. Beyond Location

However, the novelty and potential of Simjacker does not stop there. 
Retrieving a person’s location is one thing, but by using the same 
technique, and by modifying the attack message, the attacker could 
instruct the UICC to execute a range of other attacks. This is because 
using the same method the attacker has access to the complete STK 
command set, some examples of these STK commands are:

    PLAY TONE
    SEND SHORT MESSAGE
    SET UP CALL
    SEND USSD
    SEND SS
    PROVIDE LOCAL INFORMATION
        Location Information, IMEI, Battery, Network, Language, etc
    POWER OFF CARD
    RUN AT COMMAND
    SEND DTMF COMMAND
    LAUNCH BROWSER
    OPEN CHANNEL
        CS BEARER, DATA SERVICE BEARER, LOCAL BEARER, UICC SERVER MODE, etc
    SEND DATA
    GET SERVICE INFORMATION
    SUBMIT MULTIMEDIA MESSAGE
    GEOGRAPHICAL LOCATION REQUEST



By using these commands in our own tests, we were able to make targeted 
handsets open up web browsers, ring other phones, send text messages and 
so on. These attacks could be used to fulfil such purposes as

    Mis-information (e.g. by sending SMS/MMS messages with attacker controlled content)
    Fraud (e.g. by dialling premium rate numbers),
    Espionage (as well as the location retrieving attack an attacked device it could function as a listening device, by ringing a number),
    Malware spreading (by forcing a browser to open a web page with malware located on it)
    Denial of service (e.g by disabling the SIM card)
    Information retrieval (retrieve other information like language, radio type, battery level etc.)



It even may be possible to go even further - depending on handset type - 
which we will discuss in our VB2019 presentation. Worryingly, we are not 
the only people to think of these additional attacks, over the last few 
weeks and months we have observed the attackers themselves experiment 
with these different capabilities.

Finally, another benefit of Simjacker from the attacker’s perspective is 
that many of its attacks seems to work independent of handset types, as 
the vulnerability is dependent on the software on the UICC and not the 
device. We have observed devices from nearly every manufacturer being 
successfully targeted to retrieve location: Apple, ZTE, Motorola, 
Samsung, Google, Huawei, and even IoT devices with SIM cards. One 
important note is that for some specific attacks handset types do 
matter. Some, such as setting up a call, require user interaction to 
confirm, but this is not guaranteed and older phones or devices with no 
keypad or screens (such as IoT device) may not even ask for this.

```

The bottom line is that these type of attacks has been known for years, and have been used by government associated companies to develop attack software that can so all of the above, and likely much more. As time has passed eventually these secrets slowly leak out to the public.

_Originally posted by @E3V3A in https://github.com/CHEF-KOCH/NSABlocklist/issues/43#issuecomment-531601951_

**Comment by CHEF-KOCH** [comment]
Interesting find E3V3A and the reason I put this now in its own thread. The original post was under "german gov" but I think (based on the simjacker website) that this (or can) be abused around the world.

### Here is what I've found after reading the latest given infos:
- Not every SIM is affected, [AdaptiveMobileSecurity](https://www.adaptivemobile.com/blog/simjacker-next-generation-spying-over-mobile) writes on their Blog something about an S@T Browser (SIMalliance Toolbox Browser)  which is "only" been used by mobile ISPs in 30 countries. There is no mention, specifically which 30 countries. But of course the ghost is out of its bottle and the new attack could be used in the wild and be modified.
- It seems that "SIM Toolkit" is not the main problem, moreover it seems that within the context of S@T Browsers, the attacker sens something on its victims device (see graphic/video) which implants the malware. Malware then could send S@T-push-messages on the device without that the user will ever be aware of it. 
- The main problem seems that the SIM-Card allows downloading/loading the payload + execution without checking the origin of the message. 
- IMEI, device location, language settings & battery status can be transmitted to an accomplice device. It also seems that SMS/MMS could be sent from a victim's device.
- SIMAlliance  last official [publication was from 2009}](https://simalliance.org/key-technical-releases/st-specifications/st-specifications-2009/), this seems outdated, but they already reacted and [updated their specs](https://simalliance.org/key-technical-releases/st-specifications/)

## More details on this attack
- I think we can expect to see more details and migrations specific info on [Virus Bulletin](https://www.virusbulletin.com/conference/vb2019/abstracts/simjacker-next-frontier-mobile-espionage)
- The Coordinated Vulnerability Disclosure (CVD) program [GSM Association] might gives us more details after the Virus Bulletin conference 


## (Current) Migration
- Do not click on any SIM/ISP/provider popup/warning or information dialogues if you are unsure or if it looks "strange" which are send via eMail/SMS. Delete/suppress or hide such messages, if there are doubts that you are a victim of an attack, contact the experts and report your findings so that this can be debugged.
- (_needs root_) remove SIM Toolkit and SIM Toolkit + SIMappDialog (`system/app/[app1/app2]`)
- The specifications definitely needs to be updated and the Provider/SIM/protocols needs to be fixed. I don't think that (after reading the papers) Android can protect you against it, it's also not on OS problem on its own it just notifies the user.

**Comment by CHEF-KOCH** [comment]
## SIM Hijacking

[Latest study](https://www.issms2fasecure.com/assets/sim_swaps-01-10-2020.pdf) and the [latest article](https://www.techdirt.com/articles/20200114/06480143727/study-shows-internet-is-hugely-vulnerable-to-sim-hijacking-attacks.shtml), regarding this matter.


**From the paper**:
> We examined the authentication procedures used by five pre-paid wireless carriers when a customer attempted to change their SIM card. These procedures are an important line of defense against attackers who seek to hijack victims' phone numbers by posing as the victim and calling the carrier to request that service be transferred to a SIM card the attacker possesses. We found that all five carriers used insecure authentication challenges that could be easily subverted by attackers.We also found that attackers generally only needed to target the most vulnerable authentication challenges, because the rest could be bypassed.


As [Schneier mentioned](https://www.schneier.com/blog/archives/2020/01/sim_hijacking.html),  it's a trade-off between security and the usability because some people still want/need to use SMS. SMS is in some areas are entirely "free" (no extra costs, even on prepaid).
